cluster/cluster.sh

```diff
@@ -15,11 +15,27 @@
 export KUBEVIRT_PROVIDER=${KUBEVIRT_PROVIDER:-'k8s-1.17'}
 
 KUBEVIRTCI_VERSION='0b941ea5dc647d3aea6f6a6fff95563d6ce0445e'
+KUBEVIRTCI_REPO='https://github.com/kubevirt/kubevirtci.git'
 # KUBEVIRTCI_PATH var shoud be used only in vendored _kubecirtci folder
 # The CLUSTER_PATH var is used in cluster folder and points to the _kubevirtci where the cluster is deployed from.
 CLUSTER_PATH=${CLUSTER_PATH:-"${PWD}/_kubevirtci/"}
 
+function cluster::_get_repo() {
+    git --git-dir ${CLUSTER_PATH}/.git remote get-url origin
+}
+
+function cluster::_get_version() {
+    git --git-dir ${CLUSTER_PATH}/.git log --format="%H" -n 1
+}
+
 function cluster::install() {
+    # Remove cloned kubevirtci repository if it does not match the requested one
+    if [ -d ${CLUSTER_PATH} ]; then
+        if [ $(cluster::_get_repo) != ${KUBEVIRTCI_REPO} -o $(cluster::_get_version) != ${KUBEVIRTCI_VERSION} ]; then
+            rm -rf ${CLUSTER_PATH}
+        fi
+    fi
+
     if [ ! -d ${CLUSTER_PATH} ]; then
         git clone https://github.com/kubevirt/kubevirtci.git ${CLUSTER_PATH}
```

Comment:
Hey we can optimize this to get only the latest commit with adding `--branch master --depth 1`

Reply:
when I tried it I got an error:
```
+ git clone --branch master --depth 1 https://github.com/kubevirt/kubevirtci.git /home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/_kubevirtci/
Cloning into '/home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/_kubevirtci'...
remote: Enumerating objects: 1874, done.
remote: Counting objects: 100% (1874/1874), done.
remote: Compressing objects: 100% (1358/1358), done.
remote: Total 1874 (delta 565), reused 1543 (delta 445), pack-reused 0
Receiving objects: 100% (1874/1874), 3.57 MiB | 717.00 KiB/s, done.
Resolving deltas: 100% (565/565), done.
+ cd /home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/_kubevirtci/
+ git checkout 0b941ea5dc647d3aea6f6a6fff95563d6ce0445e
fatal: reference is not a tree: 0b941ea5dc647d3aea6f6a6fff95563d6ce0445e
make: *** [Makefile:123: cluster-up] Error 128

```

Reply:
Thanks for trying!